substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/hosted/SerializationFeature.java

```diff
@@ -0,0 +1,246 @@
+            try {
+                reflectionData.register(serializationTargetClass.getDeclaredConstructor((Class<?>[]) null));
+            } catch (NoSuchMethodException e) {
+                VMError.shouldNotReachHere();
+            }
+        }
+
+        reflectionData.register(serializationTargetClass);
+        boolean registerAllMethods = false;
+        boolean registerAllFieds = false;
+        try {
+            serializationTargetClass.getDeclaredField("serialVersionUID");
+        } catch (NoSuchFieldException e) {
+            registerAllMethods = true;
+            registerAllFieds = true;
+            reflectionData.register(serializationTargetClass.getDeclaredConstructors());
+        }
+
+        registerMethods(reflectionData, serializationTargetClass, registerAllMethods);
+        registerFields(reflectionData, serializationTargetClass, registerAllFieds);
+    }
+
+    private static void registerMethods(ReflectionDataBuilder reflectionData, Class<?> serializationTargetClass, boolean registerAllMethods) {
+        for (Method m : serializationTargetClass.getDeclaredMethods()) {
+            boolean register;
+            switch (m.getName()) {
+                case "readObject":
+                    register = m.getParameterCount() == 1 && m.getParameterTypes()[0].getName().equals("java.io.ObjectInputStream");
+                    break;
+                case "writeObject":
+                    register = m.getParameterCount() == 1 && m.getParameterTypes()[0].getName().equals("java.io.ObjectOutputStream");
+                    break;
+                case "readObjectNoData":
+                case "writeReplace":
+                case "readResolve":
+                    register = true;
+                    break;
+                default:
+                    register = false;
+            }
+            if (register || registerAllMethods) {
+                reflectionData.register(m);
+            }
+        }
+    }
+
+    private static void registerFields(ReflectionDataBuilder reflectionData, Class<?> serializationTargetClass, boolean registerAllFieds) {
+        int mask = Modifier.STATIC | Modifier.TRANSIENT;
+        int staticFinalMask = Modifier.STATIC | Modifier.FINAL;
+        int privateStaticFinalMask = Modifier.PRIVATE | staticFinalMask;
+
+        Set<String> serialPersistentFieldNames = new HashSet<>();
+        try {
+            Field f = serializationTargetClass.getDeclaredField("serialPersistentFields");
+            if ((f.getModifiers() & privateStaticFinalMask) == privateStaticFinalMask) {
+                f.setAccessible(true);
+                ObjectStreamField[] serialPersistentFields = (ObjectStreamField[]) f.get(null);
+                for (int i = 0; i < serialPersistentFields.length; i++) {
+                    serialPersistentFieldNames.add(serialPersistentFields[i].getName());
+                }
+            }
+        } catch (Exception e) {
```

Comment:
Please rewrite using `ReflectionUtil` and avoid such catch-all constructs.

Reply:
fixed